File: app/src/main/java/com/m7mdabaza/fooddelivery/onBoarding/FragmentAdapter.java
package com.m7mdabaza.fooddelivery.onBoarding;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.m7mdabaza.fooddelivery.onBoarding.fragments.FirstFragment;
import com.m7mdabaza.fooddelivery.onBoarding.fragments.SecondFragment;
import com.m7mdabaza.fooddelivery.onBoarding.fragments.ThirdFragment;

public class FragmentAdapter extends FragmentPagerAdapter {

    public FragmentAdapter(FragmentManager fragmentManager) {
        super(fragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        if (position == 0) {
            return new FirstFragment();
        } else if (position == 1) {
            return new SecondFragment();
        } else {
            return new ThirdFragment();
        }
    }

    @Override
    public int getCount() {
        return 3;
    }
}

File: app/src/main/java/com/m7mdabaza/fooddelivery/onBoarding/adapters/FragmentAdapter.kt
package com.m7mdabaza.fooddelivery.onBoarding.adapters

import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.m7mdabaza.fooddelivery.onBoarding.fragments.FirstFragment
import com.m7mdabaza.fooddelivery.onBoarding.fragments.SecondFragment
import com.m7mdabaza.fooddelivery.onBoarding.fragments.ThirdFragment


class FragmentAdapter(activity: AppCompatActivity, private val itemsCount: Int) :
    FragmentStateAdapter(activity) {

    override fun getItemCount(): Int {
        return itemsCount
    }

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> {
                FirstFragment()
            }
            1 -> {
                SecondFragment()
            }
            else -> {
                ThirdFragment()
            }
        }
    }


}